### PR TITLE
Background task notification

### PR DESCRIPTION
**Pull request by gregaubert**
@stas-vilchik If you want to test the licencing errors you can use the same procedure as the last PR about the license plugin. Just know that if you want the analysis to fail you have to install the commercial plugin on your instance, for that you must use the fake cobol plugin generated from the IT's of the license plugin.

### CODE REVIEW

File: server/sonar-web/src/main/js/api/ce.ts [hunk 1]
```diff
@@ -20,6 +20,31 @@
 import { getJSON, post, RequestData } from '../helpers/request';
 import throwGlobalError from '../app/utils/throwGlobalError';
 
+export interface PendingTask {
+  componentId: string;
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/api/ce.ts]: I suggest to pretend that we don't have `componentId` field 😄 It should be deprecated & removed soon

File: server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      component.configuration != undefined && component.configuration.showBackgroundTasks;
+    const url = getComponentBackgroundTaskUrl(component.key);
+    if (isInProgress) {
+      return (
+        <NavBarNotif className="alert alert-info">
+          <i className="spinner spacer-right text-bottom" />
+          <span
+            dangerouslySetInnerHTML={{
+              __html: canSeeBackgroundTasks
+                ? translateWithParameters('component_navigation.status.in_progress.admin', url)
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: I suggest using `FormattedMessage` from `react-intl` to put react components (`<Link />` in this case) inside the messages.

For example see https://github.com/SonarSource/sonarqube/pull/2584/files#diff-6a888b67f80f7a034a299b5351bfbdceR87

File: server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            )}
+          </NavBarNotif>
+        );
+      } else {
+        return (
+          <NavBarNotif className="alert alert-danger">
+            <span
+              dangerouslySetInnerHTML={{
+                __html: canSeeBackgroundTasks
+                  ? translateWithParameters('component_navigation.status.failed.admin', url)
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: blue link in red alert is super ugly

File: server/sonar-web/src/main/js/api/ce.ts (hunk 1 above)
Comment (gregaubert) [reply server/sonar-web/src/main/js/api/ce.ts]: Ah yeah good idea!

File: server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx (hunk 3 above)
Comment (gregaubert) [reply server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: Common it's not "super ugly", maybe a "bit ugly", but that's a link... Trust me, it's way better than a blue button! And we have the same colors on issues page.
I don't think we should change the color of links, do you suggest to maybe change a bit the red background ?

Comment (stas-vilchik) [reply server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: I'd suggest to use red font for link

Comment (gregaubert) [reply server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: Ok let me talk to Laura about this

Comment (gregaubert) [reply server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: Ok let me talk to Laura about this

Comment (gregaubert) [reply server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBgTaskNotif.tsx]: The problem is that red links are for dangerous/irreversible actions.

**Comment by gregaubert** [comment]
@stas-vilchik I added the fix for the sidebar and fixed your feedback expect the color link.

**Pull request merged**